include/mbedtls/md.h

```diff
@@ -49,6 +49,12 @@ typedef enum {
     MBEDTLS_MD_RIPEMD160,
 } mbedtls_md_type_t;
 
+typedef enum {
+    MBEDTLS_SIG_RSA=0,
+    MBEDTLS_SIG_ECDSA,
+    MBEDTLS_SIG_NONE,
+} mbedtls_sig_type_t;
```

Comment:
The reason not to reuse `mbedtls_pk_type_t` is that we're using this as an array index but most public key types don't have an index entry so we don't want to maintain an array that's large enough for `mbedtls_pk_type_t` values.